spec/unit/data_collector_spec.rb

```diff
@@ -39,6 +39,7 @@
   let(:new_resource) do
     new_resource = Chef::Resource::File.new("/tmp/a-file.txt")
     new_resource.checksum nil
+    new_resource.path
```

Comment:
Not sure why I need this magically this fixes all failing specs with `error_descrition` contains path attributes as well

Reply:
Without `new_resource.path` define let object `error_description` return

```
{"title"=>"Error executing action `create` on resource 'file[/tmp/a-file.txt]'", "sections"=>[{"Exception"=>"imperial to metric conversion error"}, {"Compiled Resource:"=>"# Declared in \n\nfile(\"/tmp/a-file.txt\") do\n  action [:create]\n  default_guard_interpreter :default\n  checksum nil\n  cookbook_name \"monkey\"\n  recipe_name \"atlas\"\n  owner nil\n  group nil\n  mode nil\n  verifications []\n  path \"/tmp/a-file.txt\"\nend\n"}]}
```

After `new_resource.path` seems to be expected json values

```
{"title"=>"Error executing action `create` on resource 'file[/tmp/a-file.txt]'", "sections"=>[{"Exception"=>"imperial to metric conversion error"}, {"Compiled Resource:"=>"# Declared in \n\nfile(\"/tmp/a-file.txt\") do\n  action [:create]\n  default_guard_interpreter :default\n  checksum nil\n  path \"/tmp/a-file.txt\"\n  cookbook_name \"monkey\"\n  recipe_name \"atlas\"\n  owner nil\n  group nil\n  mode nil\n  verifications []\nend\n"}]}

```

Reply:
That's due to the fact that the @path instance variable doesn't exist until the first time that `path` is called on the object where it sees that it is the name attribute so the getter pulls the value from the name attribute and sets itself.

Reply:
Ultimately there seems to be a bit of a bug in Chef::Resource#to_text around handling of the name property when it hasn't been accessed yet.